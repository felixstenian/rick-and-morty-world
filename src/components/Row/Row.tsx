import {
  FlexProps,
  GridProps,
  LayoutProps,
  ShadowProps,
  SpaceProps,
  shadow,
  space,
  system,
  flexbox,
  position,
  layout,
  color,
  ColorProps
} from 'styled-system'
import styled from 'styled-components'

export interface RowProps
  extends SpaceProps,
    LayoutProps,
    FlexProps,
    ShadowProps,
    GridProps,
    React.HTMLAttributes<HTMLDivElement>,
    Omit<ColorProps, 'color'> {
  ref?: React.ForwardedRef<unknown>
  as?: React.FC | string
  gap?: string
  cursor?: string
  wordBreak?: string
  wordWrap?: string
}

const Row: React.FC<RowProps> = styled.div<RowProps>(
  { display: 'flex' },
  system({ cursor: true, gap: true, wordWrap: true, wordBreak: true }),
  space,
  shadow,
  flexbox,
  position,
  layout,
  color
)

export default Row
