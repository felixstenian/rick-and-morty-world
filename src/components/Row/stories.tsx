import { Meta, Story } from '@storybook/react'
import Row, { RowProps } from '.'

export default {
  title: 'Row',
  component: Row,
  args: {
    children: {
      type: HTMLElement
    },
    backgroundColor: {
      type: 'string'
    },
    height: { type: 'string' },
    gap: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story<RowProps> = (args) => <Row {...args} />

Default.args = {
  children: (
    <>
      <p>Row</p>
      <p>Row</p>
      <p>Row</p>
    </>
  ),
  backgroundColor: '#2D9245',
  height: '10vh',
  gap: '10px',
  color: 'white'
}
