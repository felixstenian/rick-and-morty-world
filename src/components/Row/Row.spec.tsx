import { render, screen } from '@testing-library/react'

import Row, { RowProps } from '.'

describe('<Row />', () => {
  it('1. should render the reading', () => {
    const { container } = render(
      <Row data-testid="row">a simple description</Row>
    )

    expect(screen.getByTestId('row')).toBeInTheDocument()
    expect(screen.getByText(/a simple description/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should render a row with the correct props', () => {
    const props: RowProps = {
      gap: '10px',
      cursor: 'pointer',
      wordBreak: 'break-all',
      wordWrap: 'wrap'
    }

    const { container } = render(<Row {...props} />)

    expect(container.firstChild).toHaveStyle({
      gap: '10px',
      cursor: 'pointer',
      'word-break': 'break-all',
      'word-wrap': 'wrap'
    })
  })
})
