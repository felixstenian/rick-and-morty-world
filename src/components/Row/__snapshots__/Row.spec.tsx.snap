// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> 1. should render the reading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
  data-testid="row"
>
  a simple description
</div>
`;
